### PR TITLE
use Instant for timestamp columns

### PR DESCRIPTION
**Pull request by robfletcher**
Finally figured out how to configure this

### CODE REVIEW

File: keel-sql/src/main/resources/jooqConfig.xml [hunk 1]
```diff
@@ -20,6 +20,11 @@
           <name>LONGVARCHAR</name>
           <includeTypes>JSON</includeTypes>
         </forcedType>
+        <forcedType>
+          <userType>java.time.Instant</userType>
+          <includeTypes>TIMESTAMP</includeTypes>
+          <converter>com.netflix.spinnaker.jooq.TimestampToInstantConverter</converter>
+        </forcedType>
```
Comment (robfletcher) [inline keel-sql/src/main/resources/jooqConfig.xml:27]: Here's the bit of JOOQ configuration magic required

File: keel-sql/src/main/resources/jooqConfig.xml [hunk 2]
```diff
@@ -20,6 +20,11 @@
           <name>LONGVARCHAR</name>
           <includeTypes>JSON</includeTypes>
         </forcedType>
+        <forcedType>
+          <userType>java.time.Instant</userType>
+          <includeTypes>TIMESTAMP</includeTypes>
+          <converter>com.netflix.spinnaker.jooq.LocalDateTimeToInstantConverter</converter>
+        </forcedType>
```
Comment (robfletcher) [inline keel-sql/src/main/resources/jooqConfig.xml:27]: Here's the bit of JOOQ configuration magic required

File: keel-sql/src/main/kotlin/com/netflix/spinnaker/jooq/LocalDateTimeToInstantConverter.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+class LocalDateTimeToInstantConverter : AbstractConverter<LocalDateTime, Instant>(
+  LocalDateTime::class.java,
+  Instant::class.java
+) {
+  override fun from(databaseObject: LocalDateTime?): Instant? =
+    databaseObject?.toInstant(UTC)
+
+  override fun to(userObject: Instant?): LocalDateTime? =
+    userObject?.atZone(UTC)?.toLocalDateTime()
+}
```
Comment (robfletcher) [inline keel-sql/src/main/kotlin/com/netflix/spinnaker/jooq/LocalDateTimeToInstantConverter.kt:17]: Converter that converts between the type JOOQ wants to use by default (`LocalDateTime`) and the one we want to use (`Instant`)

File: keel-sql/src/main/resources/jooqConfig.xml (hunk 2 above)
Comment (emjburns) [reply keel-sql/src/main/resources/jooqConfig.xml:27]: LOVE

**Review by emjburns — APPROVED** [review]

**Comment by luispollo** [comment]
> Finally figured out how to configure this

So glad you did! I tried this before and couldn't make it work, but I think it was back when we had half the columns using `BIGINT` and half using `TIMESTAMP`. Thanks for taking the time to improve this!

**Pull request merged**